whole-tale/girder_wholetale
# Allow to specify a subset of dataset during import via path

### Rationale
Often during registration of a binder/dataset as a Tale we're interested only in a subset of the imported tree. This PR introduces `dsRootPath` parameter to `POST /tale/import` that allows to specify internal dataset path that should be used for generation of Tale's `dataSet`.

### How to test?
1. Import a Tale via:

```
curl 'https://girder.local.wholetale.org/api/v1/tale/import?git=false&url=https%3A%2F%2Fpbcconsortium.s3.amazonaws.com%2Fwholetale%2F5ad7cdf55b0d5007601015b7ff1ea8d6%2F2021-11-09_21.47.58%2FDataset_1-882P.zip&taleKwargs=%7B%22title%22%3A%22foo%22%7D&spawn=false&lookupKwargs=%7B%7D&imageId=6207bb1ada801d652afb4245&asTale=false&dsRootPath=%2Fdata' \
  -X 'POST' \
  -H 'girder-token: <your_token>' 
```
2. Inspect the resulting Tale. Data folder should contain `assets` and a bunch of csvs.
3. (optionally) experiment with something that's not Deriva, e.g. Dataverse ds with hierarchy. Check that selecting subpath works for both asTale=True and asTale=False.
